vnext/ReactUWP/Views/Image/ReactImage.cpp

```diff
@@ -65,51 +108,111 @@ winrt::fire_and_forget ReactImage::Source(ImageSource source) {
-              if (auto strong_this{weak_this.get()}) {
-                bool succeeded{false};
-                if (args.Status() == winrt::LoadedImageSourceLoadStatus::Success) {
-                  strong_this->m_brush->Source(surface);
-                  succeeded = true;
-                }
-                strong_this->m_onLoadEndEvent(*strong_this, succeeded);
-              }
-            });
+      if (!needsDownload || strong_this->m_memoryStream) {
+        strong_this->SetBackground(needsDownload || inlineData, true);
       }
     }
   } catch (winrt::hresult_error const &) {
-    if (auto strong_this{weak_this.get()})
+    if (auto strong_this{weak_this.get()}) {
       strong_this->m_onLoadEndEvent(*strong_this, false);
+    }
   }
-} // namespace uwp
+}
+
+void ReactImage::SetBackground(bool fromStream, bool fireLoadEndEvent) {
+  const winrt::Uri uri{Utf8ToUtf16(m_imageSource.uri)};
+
+  if (m_useCompositionBrush) {
+    const auto compositionBrush = ReactImageBrush::Create();
+    compositionBrush->AvailableSize(m_availableSize);
+    compositionBrush->ResizeMode(m_resizeMode);
+
+    auto surface = fromStream ? winrt::LoadedImageSurface::StartLoadFromStream(m_memoryStream)
+                              : winrt::LoadedImageSurface::StartLoadFromUri(uri);
+
+    m_surfaceLoadedRevoker = surface.LoadCompleted(
+        winrt::auto_revoke,
+        [weak_this{get_weak()}, compositionBrush, surface, fireLoadEndEvent](
+            winrt::LoadedImageSurface const & /*sender*/, winrt::LoadedImageSourceLoadCompletedEventArgs const &args) {
+          if (auto strong_this{weak_this.get()}) {
+            bool succeeded{false};
+            if (args.Status() == winrt::LoadedImageSourceLoadStatus::Success) {
+              winrt::Size dipsSize{surface.DecodedSize()};
+              strong_this->m_imageSource.height = dipsSize.Height;
+              strong_this->m_imageSource.width = dipsSize.Width;
+
+              compositionBrush->Source(surface);
+              strong_this->Background(compositionBrush.as<winrt::XamlCompositionBrushBase>());
+              succeeded = true;
+            }
+
+            if (fireLoadEndEvent) {
+              strong_this->m_onLoadEndEvent(*strong_this, succeeded);
+            }
+          }
+        });
+
+  } else {
+    winrt::ImageBrush bitmapBrush{};
+
+    // ImageOpened and ImageFailed are mutually exclusive. One event of the other will
+    // always fire whenever an ImageBrush has the ImageSource value set or reset.
+    m_imageOpenedRevoker = bitmapBrush.ImageOpened(
+        winrt::auto_revoke, [weak_this{get_weak()}, bitmapBrush, fireLoadEndEvent](const auto &, const auto &) {
+          if (auto strong_this{weak_this.get()}) {
+            const auto bitmap{bitmapBrush.ImageSource().as<winrt::BitmapImage>()};
+            strong_this->m_imageSource.height = bitmap.PixelHeight();
+            strong_this->m_imageSource.width = bitmap.PixelWidth();
```

Comment:
Here you're using physical pixels, and above for the LIS case you're using DIPs.  These are different units and should be consistent.  I can't tell from the docs which units are expected though.  Can you do some research on this and make the two agree?